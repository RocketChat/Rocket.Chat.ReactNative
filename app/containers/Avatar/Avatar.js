import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import FastImage from '@rocket.chat/react-native-fast-image';
import Touchable from 'react-native-platform-touchable';
import { settings as RocketChatSettings } from '@rocket.chat/sdk';

import { avatarURL } from '../../utils/avatar';
import Emoji from '../markdown/Emoji';

const Avatar = React.memo(({
	text,
	size,
	server,
	borderRadius,
	style,
	avatar,
	type,
	children,
	user,
	onPress,
	emoji,
	theme,
	getCustomEmoji,
	avatarETag,
	isStatic,
	rid
}) => {
	if ((!text && !avatar && !emoji && !rid) || !server) {
		return null;
	}

	const avatarStyle = {
		width: size,
		height: size,
		borderRadius
	};

	let image;
	if (emoji) {
		image = (
			<Emoji
				theme={theme}
				baseUrl={server}
				getCustomEmoji={getCustomEmoji}
				isMessageContainsOnlyEmoji
				literal={emoji}
				style={avatarStyle}
			/>
		);
	} else {
		let uri = avatar;
		if (!isStatic) {
			uri = avatarURL({
				type,
				text,
				size,
				user,
				avatar,
				server,
				avatarETag,
				rid
			});
		}

		image = (
			<FastImage
				style={avatarStyle}
				source={{
					uri,
					headers: RocketChatSettings.customHeaders,
					priority: FastImage.priority.high
				}}
			/>
		);
	}

	if (onPress) {
		image = (
			<Touchable onPress={onPress}>
				{image}
			</Touchable>
		);
	}


	return (
		<View style={[avatarStyle, style]}>
			{image}
			{children}
		</View>
	);
});

Avatar.propTypes = {
	server: PropTypes.string,
	style: PropTypes.any,
	text: PropTypes.string,
	avatar: PropTypes.string,
	emoji: PropTypes.string,
	size: PropTypes.number,
	borderRadius: PropTypes.number,
	type: PropTypes.string,
	children: PropTypes.object,
	user: PropTypes.shape({
		id: PropTypes.string,
		token: PropTypes.string
	}),
	theme: PropTypes.string,
	onPress: PropTypes.func,
	getCustomEmoji: PropTypes.func,
	avatarETag: PropTypes.string,
	isStatic: PropTypes.bool,
	rid: PropTypes.string
};

Avatar.defaultProps = {
	text: '',
	size: 25,
	type: 'd',
	borderRadius: 4
};

export default Avatar;
