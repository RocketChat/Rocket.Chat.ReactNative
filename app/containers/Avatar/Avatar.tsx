import React from 'react';
import { View } from 'react-native';
import FastImage from 'react-native-fast-image';
import Touchable from 'react-native-platform-touchable';
import { settings as RocketChatSettings } from '@rocket.chat/sdk';

import { getAvatarURL } from '../../lib/methods/helpers/getAvatarUrl';
import { SubscriptionType } from '../../definitions';
import Emoji from '../markdown/Emoji';
import { IAvatar } from './interfaces';

const Avatar = React.memo(
	({
		server,
		style,
		avatar,
		children,
		userId,
		token,
		onPress,
		emoji,
		getCustomEmoji,
		avatarETag,
		isStatic,
		rid,
		blockUnauthenticatedAccess,
		serverVersion,
		text,
		size = 25,
		borderRadius = 4,
		type = SubscriptionType.DIRECT,
		externalProviderUrl
	}: IAvatar) => {
		if ((!text && !avatar && !emoji && !rid) || !server) {
			return null;
		}

		const avatarStyle = {
			width: size,
			height: size,
			borderRadius
		};

		let image;
		if (emoji) {
			image = <Emoji getCustomEmoji={getCustomEmoji} isMessageContainsOnlyEmoji literal={emoji} style={avatarStyle} />;
		} else {
			let uri = avatar;
			if (!isStatic) {
				uri = getAvatarURL({
					type,
					text,
					size,
					userId,
					token,
					avatar,
					server,
					avatarETag,
					serverVersion,
					rid,
					blockUnauthenticatedAccess,
					externalProviderUrl
				});
			}

			image = (
				<FastImage
					style={avatarStyle}
					source={{
						uri,
						headers: RocketChatSettings.customHeaders,
						priority: FastImage.priority.high
					}}
				/>
			);
		}

		if (onPress) {
			image = <Touchable onPress={onPress}>{image}</Touchable>;
		}

		return (
			<View style={[avatarStyle, style]} testID='avatar'>
				{image}
				{children}
			</View>
		);
	}
);

export default Avatar;
