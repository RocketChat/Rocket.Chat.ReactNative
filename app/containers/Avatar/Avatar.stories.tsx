import React from 'react';
import { StyleSheet } from 'react-native';

import Status from '../Status/Status';
import sharedStyles from '../../views/Styles';
import Avatar from './Avatar';

const styles = StyleSheet.create({
	custom: {
		padding: 16
	}
});

const server = 'https://open.rocket.chat';

export default {
	title: 'Avatar'
};

export const AvatarText = () => <Avatar text='Avatar' server={server} size={56} />;

export const AvatarRoomId = () => <Avatar type='p' rid='devWBbYr7inwupPqK' server={server} size={56} />;

export const AvatarUrl = () => (
	<Avatar
		avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
		server={server}
		size={56}
	/>
);

export const AvatarPath = () => <Avatar avatar='/avatar/diego.mello' server={server} size={56} />;

export const WithETag = () => (
	<Avatar type='d' text='djorkaeff.alexandre' avatarETag='5ag8KffJcZj9m5rCv' server={server} size={56} />
);

export const WithoutETag = () => <Avatar type='d' text='djorkaeff.alexandre' server={server} size={56} />;

export const Emoji = () => (
	<Avatar emoji='troll' getCustomEmoji={() => ({ name: 'troll', extension: 'jpg' })} server={server} size={56} />
);

export const Direct = () => <Avatar text='diego.mello' server={server} type='d' size={56} />;

export const Channel = () => <Avatar text='general' server={server} type='c' size={56} />;

export const Touchable = () => <Avatar text='Avatar' server={server} onPress={() => console.log('Pressed!')} size={56} />;

export const Static = () => (
	<Avatar
		avatar='https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg'
		server={server}
		isStatic
		size={56}
	/>
);

export const CustomBorderRadius = () => <Avatar text='Avatar' server={server} borderRadius={28} size={56} />;

export const Children = () => (
	<Avatar text='Avatar' server={server} size={56}>
		<Status size={24} style={[sharedStyles.status]} status='busy' />
	</Avatar>
);

export const WrongServer = () => <Avatar text='Avatar' server='https://google.com' size={56} />;

export const CustomStyle = () => <Avatar text='Avatar' server={server} size={56} style={styles.custom} />;
