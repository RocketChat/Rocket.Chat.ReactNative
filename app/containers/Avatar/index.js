import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Q } from '@nozbe/watermelondb';

import database from '../../lib/database';
import { getUserSelector } from '../../selectors/login';
import Avatar from './Avatar';

class AvatarContainer extends React.Component {
	static propTypes = {
		rid: PropTypes.string,
		text: PropTypes.string,
		type: PropTypes.string,
		blockUnauthenticatedAccess: PropTypes.bool,
		serverVersion: PropTypes.string
	};

	static defaultProps = {
		text: '',
		type: 'd'
	};

	constructor(props) {
		super(props);
		this.mounted = false;
		this.state = { avatarETag: '' };
		this.init();
	}

	componentDidMount() {
		this.mounted = true;
	}

	componentDidUpdate(prevProps) {
		const { text, type } = this.props;
		if (prevProps.text !== text || prevProps.type !== type) {
			this.init();
		}
	}

	componentWillUnmount() {
		if (this.subscription?.unsubscribe) {
			this.subscription.unsubscribe();
		}
	}

	get isDirect() {
		const { type } = this.props;
		return type === 'd';
	}

	init = async() => {
		const db = database.active;
		const usersCollection = db.get('users');
		const subsCollection = db.get('subscriptions');

		let record;
		try {
			if (this.isDirect) {
				const { text } = this.props;
				const [user] = await usersCollection.query(Q.where('username', text)).fetch();
				record = user;
			} else {
				const { rid } = this.props;
				record = await subsCollection.find(rid);
			}
		} catch {
			// Record not found
		}

		if (record) {
			const observable = record.observe();
			this.subscription = observable.subscribe((r) => {
				const { avatarETag } = r;
				if (this.mounted) {
					this.setState({ avatarETag });
				} else {
					this.state.avatarETag = avatarETag;
				}
			});
		}
	}

	render() {
		const { avatarETag } = this.state;
		const { serverVersion } = this.props;
		return (
			<Avatar
				avatarETag={avatarETag}
				serverVersion={serverVersion}
				{...this.props}
			/>
		);
	}
}

const mapStateToProps = state => ({
	user: getUserSelector(state),
	server: state.share.server.server || state.server.server,
	serverVersion: state.share.server.version || state.server.version,
	blockUnauthenticatedAccess:
		state.share.settings?.Accounts_AvatarBlockUnauthenticatedAccess
		?? state.settings.Accounts_AvatarBlockUnauthenticatedAccess
		?? true
});
export default connect(mapStateToProps)(AvatarContainer);
