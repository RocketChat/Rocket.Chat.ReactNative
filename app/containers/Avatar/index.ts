import Avatar from './AvatarContainer';

export { default as AvatarWithEdit } from './AvatarWithEdit';

export default Avatar;
