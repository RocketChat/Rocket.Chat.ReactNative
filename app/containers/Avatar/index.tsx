import React from 'react';
import { connect } from 'react-redux';
import { Q } from '@nozbe/watermelondb';
import { Observable, Subscription } from 'rxjs';

import database from '../../lib/database';
import { getUserSelector } from '../../selectors/login';
import { IApplicationState, TSubscriptionModel, TUserModel } from '../../definitions';
import Avatar from './Avatar';
import { IAvatar } from './interfaces';

class AvatarContainer extends React.Component<IAvatar, any> {
	private subscription?: Subscription;

	static defaultProps = {
		text: '',
		type: 'd'
	};

	constructor(props: IAvatar) {
		super(props);
		this.state = { avatarETag: '' };
		this.init();
	}

	componentDidUpdate(prevProps: IAvatar) {
		const { text, type } = this.props;
		if (prevProps.text !== text || prevProps.type !== type) {
			this.init();
		}
	}

	shouldComponentUpdate(nextProps: IAvatar, nextState: { avatarETag: string }) {
		const { avatarETag } = this.state;
		const { text, type, size, externalProviderUrl } = this.props;
		if (nextProps.externalProviderUrl !== externalProviderUrl) {
			return true;
		}
		if (nextState.avatarETag !== avatarETag) {
			return true;
		}
		if (nextProps.text !== text) {
			return true;
		}
		if (nextProps.type !== type) {
			return true;
		}
		if (nextProps.size !== size) {
			return true;
		}

		return false;
	}

	componentWillUnmount() {
		if (this.subscription?.unsubscribe) {
			this.subscription.unsubscribe();
		}
	}

	get isDirect() {
		const { type } = this.props;
		return type === 'd';
	}

	init = async () => {
		const db = database.active;
		const usersCollection = db.get('users');
		const subsCollection = db.get('subscriptions');

		let record;
		try {
			if (this.isDirect) {
				const { text } = this.props;
				const [user] = await usersCollection.query(Q.where('username', text)).fetch();
				record = user;
			} else {
				const { rid } = this.props;
				if (rid) {
					record = await subsCollection.find(rid);
				}
			}
		} catch {
			// Record not found
		}

		if (record) {
			const observable = record.observe() as Observable<TSubscriptionModel | TUserModel>;
			this.subscription = observable.subscribe(r => {
				const { avatarETag } = r;
				this.setState({ avatarETag });
			});
		}
	};

	render() {
		const { avatarETag } = this.state;
		const { serverVersion } = this.props;
		return <Avatar {...this.props} avatarETag={avatarETag} serverVersion={serverVersion} />;
	}
}

const mapStateToProps = (state: IApplicationState) => ({
	user: getUserSelector(state),
	server: state.share.server.server || state.server.server,
	serverVersion: state.share.server.version || state.server.version,
	blockUnauthenticatedAccess:
		(state.share.settings?.Accounts_AvatarBlockUnauthenticatedAccess as boolean) ??
		state.settings.Accounts_AvatarBlockUnauthenticatedAccess ??
		true,
	externalProviderUrl: state.settings.Accounts_AvatarExternalProviderUrl as string
});
export default connect(mapStateToProps)(AvatarContainer);
