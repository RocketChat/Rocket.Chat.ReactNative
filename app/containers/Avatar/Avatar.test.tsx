import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './Avatar.stories';

generateSnapshots(stories);
