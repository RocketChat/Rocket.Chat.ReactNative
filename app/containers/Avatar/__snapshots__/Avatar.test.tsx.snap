// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: AvatarExternalProviderUrl should match snapshot 1`] = `
<View
  accessibilityLabel="Avatar's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://hips.hearstapps.com/hmg-prod/images/dog-puppy-on-garden-royalty-free-image-1586966191.jpg?crop=0.752xw:1.00xh;0.175xw,0&resize=1200:*&format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: AvatarPath should match snapshot 1`] = `
<View
  accessibilityLabel="[missing {{username}} value]'s avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/diego.mello?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: AvatarRoomId should match snapshot 1`] = `
<View
  accessibilityLabel="[missing {{username}} value]'s avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/room/devWBbYr7inwupPqK?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: AvatarText should match snapshot 1`] = `
<View
  accessibilityLabel="Avatar's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/Avatar?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: AvatarUrl should match snapshot 1`] = `
<View
  accessibilityLabel="[missing {{username}} value]'s avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: Channel should match snapshot 1`] = `
<View
  accessibilityLabel="general's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/@general?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: Children should match snapshot 1`] = `
<View
  accessibilityLabel="Avatar's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/Avatar?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#D40C26",
          "fontSize": 48,
        },
        [
          {
            "lineHeight": 48,
          },
          [
            {
              "height": 48,
              "textAlignVertical": "center",
              "width": 48,
            },
            [
              {
                "borderRadius": 10,
                "bottom": -2,
                "position": "absolute",
                "right": -2,
              },
            ],
          ],
        ],
        {
          "fontFamily": "custom",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î¥ž
  </Text>
</View>
`;

exports[`Story Snapshots: CustomBorderRadius should match snapshot 1`] = `
<View
  accessibilityLabel="Avatar's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 28,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={28}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/Avatar?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 28,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: CustomStyle should match snapshot 1`] = `
<View
  accessibilityLabel="Avatar's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      {
        "padding": 16,
      },
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/Avatar?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: Direct should match snapshot 1`] = `
<View
  accessibilityLabel="diego.mello's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/diego.mello?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: Emoji should match snapshot 1`] = `
<View
  accessibilityLabel="[missing {{username}} value]'s avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="contain"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "uri": "https://open.rocket.chat/emoji-custom/troll.jpg",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: RoomAvatarExternalProviderUrl should match snapshot 1`] = `
<View
  accessibilityLabel="[missing {{username}} value]'s avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://cdn.pensador.com/img/authors/ho/me/homer-simpson-l.jpg?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: Static should match snapshot 1`] = `
<View
  accessibilityLabel="[missing {{username}} value]'s avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://user-images.githubusercontent.com/29778115/89444446-14738480-d728-11ea-9412-75fd978d95fb.jpg",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: Touchable should match snapshot 1`] = `
<View
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <View
    accessibilityLabel="Avatar's avatar"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      borderRadius={4}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={56}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={[]}
      priority="high"
      source={
        [
          {
            "headers": {
              "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
            },
            "uri": "https://open.rocket.chat/avatar/Avatar?format=png&size=112",
          },
        ]
      }
      style={
        {
          "borderRadius": 4,
          "height": 56,
          "width": 56,
        }
      }
      transition={null}
      width={56}
    />
  </View>
</View>
`;

exports[`Story Snapshots: WithETag should match snapshot 1`] = `
<View
  accessibilityLabel="djorkaeff.alexandre's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/djorkaeff.alexandre?format=png&size=112&etag=5ag8KffJcZj9m5rCv",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: WithoutETag should match snapshot 1`] = `
<View
  accessibilityLabel="djorkaeff.alexandre's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://open.rocket.chat/avatar/djorkaeff.alexandre?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;

exports[`Story Snapshots: WrongServer should match snapshot 1`] = `
<View
  accessibilityLabel="Avatar's avatar"
  accessible={true}
  style={
    [
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      },
      undefined,
    ]
  }
  testID="avatar"
>
  <ViewManagerAdapter_ExpoImage
    borderRadius={4}
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={56}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    priority="high"
    source={
      [
        {
          "headers": {
            "User-Agent": "RC Mobile; ios unknown; vunknown (unknown)",
          },
          "uri": "https://google.com/avatar/Avatar?format=png&size=112",
        },
      ]
    }
    style={
      {
        "borderRadius": 4,
        "height": 56,
        "width": 56,
      }
    }
    transition={null}
    width={56}
  />
</View>
`;
